compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -652,10 +652,14 @@ impl<'tcx> TypeVisitor<'tcx> for OrphanChecker<'tcx> {
             | ty::Slice(..)
             | ty::RawPtr(..)
             | ty::Never
-            | ty::Tuple(..)
-            | ty::Projection(..) => self.found_non_local_ty(ty),
+            | ty::Tuple(..) => self.found_non_local_ty(ty),
 
-            ty::Param(..) => self.found_param_ty(ty),
+            ty::Param(..) => self.found_uncovered_ty(ty),
+
+            ty::Projection(..) => {
+                self.found_non_local_ty(ty);
```

Comment:
i think we want to only add projections as uncovered types and never as non local ones